File: 코딩테스트 입문/영어/OX퀴즈.java
class Solution {
    public String[] solution(String[] quiz) {
        String[] answer = new String[quiz.length];

        for (int i = 0; i < quiz.length; i++) {
            String[] parts = quiz[i].split(" ");
            int X = Integer.parseInt(parts[0]);
            String operator = parts[1];
            int Y = Integer.parseInt(parts[2]);
            int Z = Integer.parseInt(parts[4]);

            int result = 0;
            if (operator.equals("+")) {
                result = X + Y;
            } else if (operator.equals("-")) {
                result = X - Y;
            }

            if (result == Z) {
                answer[i] = "O";
            } else {
                answer[i] = "X";
            }
        }
        return answer;
    }
}

File: 코딩테스트 입문/OX퀴즈.cpp
#include <string>
#include <vector>
#include <sstream>

using namespace std;

vector<string> solution(vector<string> quiz) {
    vector<string> answer;

    for (auto q : quiz) {
        stringstream ss(q);
        int X, Y, Z;
        string operator_sign, equal_sign;

        ss >> X >> operator_sign >> Y >> equal_sign >> Z;

        int result = 0;
        if (operator_sign == "+") {
            result = X + Y;
        } else if (operator_sign == "-") {
            result = X - Y;
        }

        if (result == Z) {
            answer.push_back("O");
        } else {
            answer.push_back("X");
        }
    }

    return answer;
}